### PR TITLE
Converts ClearBrowsingDataPanel into redux component

### LINKED ISSUE #9440: Redux component - ClearBrowsingDataPanel
## Test plan

https://github.com/brave/browser-laptop/pull/9457#issue-235903786

----

**Describe the issue you encountered:**
Convert `ClearBrowsingDataPanel` into redux component.

### PR DESCRIPTION
**Pull request by NejcZdovc**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Resolves #9440

Auditors: @bridiver @bsclifton

Test Plan:
- check if you can clear history
- check if toggles are preserved as you click clear
- toggle one switch, click cancel and check that toggle was not preserved when you open it next time
- check if ledger is still working correctly: ledger initi, ledger backup, ledger recovery

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/ledger.js [hunk 1]
```diff
@@ -168,7 +168,8 @@ const doAction = (action) => {
       break
 
     case appConstants.APP_ON_CLEAR_BROWSING_DATA:
-      if (action.clearDataDetail.get('browserHistory') && !getSetting(settings.PAYMENTS_ENABLED)) reset(true)
+      // TODO
```
Comment (NejcZdovc) [inline app/ledger.js]: @bsclifton @bridiver I would need your help here. What we need to add here is state (now we only get action) so that you can check it the same way we do it for sites. How should we address this problem?

**Comment by NejcZdovc** [comment]
Except this ledger problem everything else is done.

File: app/ledger.js (hunk 1 above)
Comment (NejcZdovc) [inline app/ledger.js]: I was thinking to add state directly in same as we do with other reducers. One thing that we can also do is get state only in this case with `getState`

**Comment by NejcZdovc** [comment]
Don't squash commits, because second one is quite special and I would keep it separated.

File: app/browser/menu.js [hunk 2]
```diff
@@ -693,11 +693,11 @@ const doAction = (action) => {
       }
       break
     case appConstants.APP_ON_CLEAR_BROWSING_DATA:
-      if (action.clearDataDetail.get('browserHistory')) {
-        appDispatcher.waitFor([appStore.dispatchToken], () => {
+      appDispatcher.waitFor([appStore.dispatchToken], () => {
+        if (appStore.getState().getIn(['clearBrowsingDataDefaults', 'browserHistory'])) {
```
Comment (bridiver) [inline app/browser/menu.js]: why the change here from action.clearDataDetail to checking the state?

Comment (NejcZdovc) [inline app/browser/menu.js]: Because we are not sending anything in the action anymore. We have everything that we need directly in the store already. We are saving data here already https://github.com/brave/browser-laptop/pull/9457/files#diff-23ca389e2bcb77191b5a9c10900eb3a3R697

Comment (NejcZdovc) [inline app/browser/menu.js]: actions has no params https://github.com/brave/browser-laptop/pull/9457/files#diff-5fea3af223318beea03d50c8ee3de64dR591
